### PR TITLE
feat: native token config

### PR DESCRIPTION
**Pull request by r4mmer**
### Acceptance Criteria

- Remove all uses of `HATHOR_TOKEN_CONFIG` in favor of `NATIVE_TOKEN_UID` and the configured native token.
- Start the wallet by adding the native token received from the fullnode

Requires https://github.com/HathorNetwork/hathor-wallet-lib/pull/689 to be merged and released.

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/sagas/wallet.js [hunk 1]
```diff
@@ -250,6 +252,13 @@ export function* startWallet(action) {
     });
     console.log('[+] Start wallet.', serverInfo);
 
+    // Use HTR as default
+    let nativeToken = hathorLibConstants.DEFAULT_TOKEN_CONFIG;
+    if (info.native_token) {
+      nativeToken = info.native_token;
+    }
```
Comment (pedroferreira1) [inline src/sagas/wallet.js:259]: Can we use `wallet.storage.getNativeTokenData()`?

File: src/sagas/wallet.js [hunk 2]
```diff
@@ -250,6 +252,13 @@ export function* startWallet(action) {
     });
     console.log('[+] Start wallet.', serverInfo);
 
+    // Use HTR as default
+    let nativeToken = hathorLibConstants.DEFAULT_TOKEN_CONFIG;
```
Comment (pedroferreira1) [inline src/sagas/wallet.js:256]: If we can't use the method from storage, we should fix this, to use `DEFAULT_NATIVE_TOKEN_CONFIG`

File: src/screens/CustomTokens.js [hunk 3]
```diff
@@ -72,7 +72,7 @@ function CustomTokens() {
     <div className="content-wrapper">
       <BackButton />
       <h3 className="mt-4">{t`Custom Tokens`}</h3>
-      <p className="mt-5">{t`You can create your own digital token with customized specifications on Hathor Network with only a few clicks. They will fully work under the same technical assumptions of high scalability and decentralized consensus of our native HTR tokens. Custom tokens will always work independently of the price of the native HTR token, and they can serve multiple purposes.`}</p>
```
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/screens/CustomTokens.js:75]: Done

File: src/sagas/wallet.js (hunk 1 above)
Comment (r4mmer) [reply src/sagas/wallet.js]: Done

File: src/reducers/index.js [hunk 4]
```diff
@@ -1012,4 +1005,29 @@ export const onSetMiningServer = (state, { payload }) => ({
...
+  if (state.tokens.length === 0) {
+    tokens = [{...payload, uid: NATIVE_TOKEN_UID}];
+  } else {
+    const nativeTokenIndex = findIndex(state.tokens, (e) => e.uid === NATIVE_TOKEN_UID);
+    tokens[nativeTokenIndex] = {...payload, uid: NATIVE_TOKEN_UID};
+  }
+
+  return {
+    ...state,
+    nativeTokenData: payload,
```
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/reducers/index.js:1024]: Done

File: src/reducers/index.js [hunk 5]
```diff
@@ -1012,4 +1005,29 @@ export const onSetMiningServer = (state, { payload }) => ({
   miningServer: payload,
 });
 
+export const onSetNativeTokenData = (state, { payload }) => {
+  let tokens = [...state.tokens];
+  if (state.tokens.length === 0) {
+    tokens = [{...payload, uid: NATIVE_TOKEN_UID}];
+  } else {
+    const nativeTokenIndex = findIndex(state.tokens, (e) => e.uid === NATIVE_TOKEN_UID);
```
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/reducers/index.js:1014]: Added a case for `findIndex` returning `-1` which means the index was not found.
I don't think findIndex can return null

File: src/actions/index.js [hunk 6]
```diff
@@ -502,3 +503,8 @@ export const setMiningServer = (url) => ({
   type: types.SET_MINING_SERVER,
   payload: url,
 });
+
+export const setNativeTokenData = (data) => ({
```
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/actions/index.js:510]: Done

File: src/components/tokens/TokenMint.js [hunk 7]
```diff
@@ -210,13 +210,14 @@ class TokenMint extends React.Component {
...
     const depositPercent = wallet.storage.getTokenDepositPercentage();
+    const nativeTokenConfig = wallet.storage.getNativeTokenData();
 
     return (
       <TokenAction
        renderForm={renderForm}
        title={t`Mint tokens`}
-       subtitle={`A deposit of ${depositPercent * 100}% in HTR of the mint amount is required`}
-       deposit={`Deposit: ${tokens.getDepositAmount(getAmountToCalculateDeposit(), depositPercent)} HTR (${hathorLib.numberUtils.prettyValue(this.props.htrBalance)} HTR available)`}
+       subtitle={`A deposit of ${depositPercent * 100}% in ${nativeTokenConfig.symbol} of the mint amount is required`}
```
Comment (tuliomir) [inline src/components/tokens/TokenMint.js:219]: ⭐ Nice idea to change the symbols also in the texts!

File: src/reducers/index.js [hunk 8]
```diff
@@ -1012,4 +1006,33 @@ export const onSetMiningServer = (state, { payload }) => ({
   miningServer: payload,
 });
 
+export const onSetNativeTokenData = (state, { payload }) => {
```
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/reducers/index.js:1009]: Done.

File: src/reducers/index.js [hunk 9]
```diff
@@ -1012,4 +1006,33 @@ export const onSetMiningServer = (state, { payload }) => ({
...
+  let tokens = [...state.tokens];
+  if (state.tokens.length === 0) {
+    tokens = [{...payload, uid: NATIVE_TOKEN_UID}];
+  } else {
+    const nativeTokenIndex = findIndex(state.tokens, (e) => e.uid === NATIVE_TOKEN_UID);
+    if (nativeTokenIndex === -1) {
+      // In case the native token is not in the list, we add it as the first token
+      tokens = [{...payload, uid: NATIVE_TOKEN_UID}, ...state.tokens];
+    }
+    tokens[nativeTokenIndex] = {...payload, uid: NATIVE_TOKEN_UID};
```
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/reducers/index.js:1019]: Nice catch, changed the organization to be more clear and fixed this issue

File: src/actions/index.js [hunk 10]
```diff
@@ -502,3 +503,11 @@ export const setMiningServer = (url) => ({
   type: types.SET_MINING_SERVER,
   payload: url,
 });
+
+/**
+ * Set the native token data
```
Comment (pedroferreira1) [inline src/actions/index.js:508]: Missing expected data type

**Review by pedroferreira1 — APPROVED** [review]

File: src/actions/index.js (hunk 10 above)
Comment (r4mmer) [reply src/actions/index.js:508]: Done

**Review by tuliomir — APPROVED** [review]

**Pull request merged**